cranelift/codegen/src/isa/unwind/systemv.rs

```diff
@@ -262,23 +263,27 @@ impl<'a, Reg: PartialEq + Copy> InstructionBuilder<'a, Reg> {
     }
 
     fn set_cfa_reg(&mut self, offset: u32, reg: Reg) -> Result<(), RegisterMappingError> {
-        self.instructions.push((
-            offset,
-            CallFrameInstruction::CfaRegister(self.map_reg.map(reg)?),
-        ));
-        self.frame_register = Some(reg);
+        match self.frame_register {
+            None => {
+                self.instructions.push((
+                    offset,
+                    CallFrameInstruction::CfaRegister(self.map_reg.map(reg)?),
+                ));
+                self.frame_register = Some(reg);
+            }
+            Some(_) => {
+                // Restore SP and its offset.
+                self.instructions.push((
+                    offset,
+                    CallFrameInstruction::Cfa(self.map_reg.map(reg)?, self.sp_offset),
+                ));
+                self.frame_register = None;
```

Comment:
I'm also having a hard time understanding why this is necessary from what it was previously implemented, namely recording a CFA change for the restore of `rbp` (i.e. in `restore_reg`) rather than having a strange, to me at least, "set frame pointer to `rsp`" unwind operation with these changes that causes the call to `set_cfa_reg`.

Yury, is there a semantic difference to the resulting baseline changes?

specifically:

```
DW_CFA_def_cfa(rsp, 8)
DW_CFA_same_value(rbp)
```

and 

```
DW_CFA_same_value (rbp)
DW_CFA_def_cfa(rsp, 8)
```

Reply:
`DW_CFA_def_cfa(rsp, 8)` and `DW_CFA_same_value(rbp)` affect different states, so in this case order is not relevant. 

I'll add `ResetFramePointer` to restore CFA to its default value, as specified as the CIE.

Reply:
A different unwind operation (e.g. `RestoreFramePointer`) I think makes sense to me 👍